src/Collector/TemplateRenderCollector.php

```diff
@@ -0,0 +1,239 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Efabrica\PHPStanLatte\Collector;
+
+use Efabrica\PHPStanLatte\Collector\ValueObject\CollectedTemplateRender;
+use Efabrica\PHPStanLatte\Resolver\NameResolver\NameResolver;
+use Efabrica\PHPStanLatte\Resolver\TypeResolver\TemplateTypeResolver;
+use Efabrica\PHPStanLatte\Resolver\ValueResolver\PathResolver;
+use Efabrica\PHPStanLatte\Resolver\ValueResolver\ValueResolver;
+use Efabrica\PHPStanLatte\Template\Variable;
+use PhpParser\Node;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Expr\BinaryOp\Plus;
+use PhpParser\Node\Expr\MethodCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Collectors\CollectedData;
+use PHPStan\Collectors\Collector;
+use PHPStan\PhpDoc\TypeStringResolver;
+use PHPStan\Type\Constant\ConstantArrayType;
+use PHPStan\Type\Constant\ConstantStringType;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\ThisType;
+
+/**
+ * @phpstan-import-type CollectedTemplateRenderArray from CollectedTemplateRender
+ * @implements Collector<MethodCall, ?CollectedTemplateRenderArray[]>
+ */
+final class TemplateRenderCollector implements Collector
+{
+    private NameResolver $nameResolver;
+
+    private ValueResolver $valueResolver;
+
+    private PathResolver $pathResolver;
+
+    private TypeStringResolver $typeStringResolver;
+
+    private TemplateTypeResolver $templateTypeResolver;
+
+    public function __construct(
+        NameResolver $nameResolver,
+        ValueResolver $valueResolver,
+        PathResolver $pathResolver,
+        TypeStringResolver $typeStringResolver,
+        TemplateTypeResolver $templateTypeResolver
+    ) {
+        $this->nameResolver = $nameResolver;
+        $this->valueResolver = $valueResolver;
+        $this->pathResolver = $pathResolver;
+        $this->typeStringResolver = $typeStringResolver;
+        $this->templateTypeResolver = $templateTypeResolver;
+    }
+
+    public function getNodeType(): string
+    {
+        return MethodCall::class;
+    }
+
+    /**
+     * @param MethodCall $node
+     * @phpstan-return null|CollectedTemplateRenderArray[]
+     */
+    public function processNode(Node $node, Scope $scope): ?array
+    {
+        $classReflection = $scope->getClassReflection();
+        if ($classReflection === null) {
+            return null;
+        }
+
+        $functionName = $scope->getFunctionName();
+        if ($functionName === null) {
+            return null;
+        }
+
+        $currentClassType = new ObjectType($classReflection->getName());
+
+        if ($currentClassType->isInstanceOf('Latte\Runtime\Template')->yes()) {
```

Comment:
Detects collector context - could be in future extended to more contexts that triggers render of template (like currently unsupported call to sendTempalte($file) in Presenters